packages/mobile/src/screens/explore-screen/SearchExploreScreen.tsx

```diff
@@ -43,144 +59,242 @@ const tiles = [
   REMIXABLES
 ]
 
+const itemKindByCategory: Record<SearchCategory, Kind | null> = {
+  all: null,
+  users: Kind.USERS,
+  tracks: Kind.TRACKS,
+  playlists: Kind.COLLECTIONS,
+  albums: Kind.COLLECTIONS
+}
+
+const { getSearchHistory } = searchSelectors
+
 export const SearchExploreScreen = () => {
   const { spacing } = useTheme()
-
+  const { params } = useRoute<'Search'>()
   const { drawerHelpers } = useContext(AppDrawerContext)
+  const isUSDCPurchasesEnabled = useIsUSDCEnabled()
 
-  const handleOpenLeftNavDrawer = useCallback(() => {
-    drawerHelpers?.openDrawer()
-  }, [drawerHelpers])
+  // State
+  const [category, setCategory] = useState<SearchCategory>(
+    params?.category ?? 'all'
+  )
+  const [filters, setFilters] = useState<SearchFiltersType>(
+    params?.filters ?? {}
+  )
+  const [bpmType, setBpmType] = useState<'range' | 'target'>('range')
+  const [autoFocus, setAutoFocus] = useState(params?.autoFocus ?? false)
+  const [searchInput, setSearchInput] = useState(params?.query ?? '')
+  const [debouncedQuery, setDebouncedQuery] = useState(searchInput)
+  useDebounce(
+    () => {
+      setDebouncedQuery(searchInput)
+    },
+    400, // debounce delay in ms
+    [searchInput]
+  )
 
-  const isUSDCPurchasesEnabled = useIsUSDCEnabled()
+  useEffect(() => {
```

Comment:
Since we are already setting the default values do we need this effect?

Reply:
oh true